Training/Contexts/AppDBContext.cs

```diff
@@ -24,5 +29,29 @@ public async Task<T> UpdateEntity<T>(T entity) where T : BaseEntity {
             await SaveChangesAsync();
             return entity;
         }
+
+        public async Task<T> GetEntityById<T>(Guid id) where T : BaseEntity 
+        {
+            T? entity = await Set<T>().FindAsync(id);
+            return entity == null ? throw new InvalidOperationException("No entity to update") : entity;
```

Comment:
dont throw exceptions return the null objects because the exception handling middleware will return an internal server error instead of bad request